x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/job/FeatureIndexBuilderJobConfig.java

```diff
@@ -6,48 +6,48 @@
 
 package org.elasticsearch.xpack.ml.featureindexbuilder.job;
 
+import org.elasticsearch.common.Nullable;
 import org.elasticsearch.common.ParseField;
 import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.io.stream.NamedWriteable;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
-import org.elasticsearch.common.io.stream.Writeable;
-import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
 import org.elasticsearch.common.xcontent.ToXContentObject;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentParser;
-
 import java.io.IOException;
 import java.util.Objects;
 
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
 /**
  * This class holds the configuration details of a feature index builder job
  */
 public class FeatureIndexBuilderJobConfig implements NamedWriteable, ToXContentObject {
-    private static final String NAME = "xpack/feature_index_builder/jobconfig";
 
-    public static final ParseField ID = new ParseField("id");
-
-    private String id;
+    private static final String NAME = "xpack/feature_index_builder/jobconfig";
+    private static final String ID = "id";
```

Comment:
Why did not remove this from being a `ParseField`? This seems to go against the prevailing pattern.

Reply:
no strong feeling, I saw both versions in code